pkg/modifiers/modifiers.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package modifiers provides the functionality to parse meta modifiers from text.
+package modifiers
+
+import (
+	"context"
+	"regexp"
+)
+
+const (
+	MetaKeyGuardianDir = "GUARDIAN_DIR"
+)
+
+var (
+	// newline is a regexp to split strings at line breaks.
+	newline = regexp.MustCompile("\r?\n")
+
+	// metaValuesKVRegex is a regex to split key value pairs for comment modifiers.
+	metaValuesKVRegex = regexp.MustCompile(`^(GUARDIAN_[A-Z0-9_]+)=(.*)$`)
+)
+
+// MetaValues is a map of key value pairs to store meta values.
+type MetaValues map[string][]string
+
+// ParseBodyMetaValues parses the modifiders from a contents string.
+func ParseBodyMetaValues(ctx context.Context, contents string) MetaValues {
```

Comment:
This is fine, but I do wonder how much this is going to buy us in the long run versus a simple `if strings.Contains("GUARDIAN_DIR")`. Seems like a lot of complexity based on the intention that we _might_ need more modifiers in the future?

Reply:
Yea, but this code existed so i just pasted it back in. The effort was very minimal.